source/s2n/s2n_tls_channel_handler.c

```diff
@@ -168,23 +168,30 @@ static const char *s_determine_default_pki_ca_file(void) {
     return NULL;
 }
 
+/* If s2n is already initialized, then we don't call s2n_init() or s2n_cleanup() ourselves */
+static bool s_s2n_initialized_externally;
+
 void aws_tls_init_static_state(struct aws_allocator *alloc) {
     (void)alloc;
     AWS_LOGF_INFO(AWS_LS_IO_TLS, "static: Initializing TLS using s2n.");
 
-    setenv("S2N_ENABLE_CLIENT_MODE", "1", 1);
-    setenv("S2N_DONT_MLOCK", "1", 1);
-
     /* Disable atexit behavior, so that s2n_cleanup() fully cleans things up.
      *
      * By default, s2n uses an ataexit handler and doesn't fully clean up until the program exits.
      * This can cause a crash if s2n is compiled into a shared library and
      * that library is unloaded before the appexit handler runs. */
-    s2n_disable_atexit();
+    if (s2n_disable_atexit() != S2N_SUCCESS) {
```

Comment:
This assumes that the only reason why `s2n_disable_atexit` would fail is that s2n is already initialized. This is correct now, but risks regression imho.

I would explicitly call out/document this assumption, with a link to the source of assumption. E.g. [link](https://code.amazon.com/packages/S2N/blobs/f9dd19fdc78e2d4d95a7b8f6326c3d84ed61d40b/--/third-party-src/api/s2n.h#L211-L212)